Task3

Geratoptus opened the pull request.

Geratoptus committed:
Рефакторинг

1) Унифицировал скобочную нотацию
2) Сменил нотацию полей с m_ на this
3) Убрал generateMenuBar, вынеся MenuBar в отдельный класс
4) Постарался разбить энтерами логические блоки кода

Geratoptus committed:
Закрытие с подтверждением

1) Добавил метод confirmWindowClose для выхода с подтверждением
2) Распространил метод confirmWindowClose на закрытие по "X" и по кнопке в меню.
3) Локализовал yesButtonText и noButtontext

Geratoptus committed:
Небольшие изменения

1) Сделал положение и размер окон параметрами
2) Сделал окна полями
3) Заменил System.exit(0) на dispose()

Geratoptus committed:
Добавил последовательный dispose всех окон

Geratoptus committed:
Небольшие изменения

1) Вместо параметров в createGameWindow и createLogWindow сделал декоратор над addWindow с параметрами.
2) Добавил ускоритель на кнопки

Geratoptus committed:
task2

Сделал сохранение состояния окон и его подгрузку из логов.

Geratoptus committed:
Добавил запоминание свернутости окон

Geratoptus committed:
Доработки

1) Сделал прослойку в виде базового класса SaveableWindow
2) Убрал класс StateManager за ненадобностью
3) Добавил кнопку для сохранения
4) Сделал подтверждение сохранения при выходе из приложения

Geratoptus committed:
Вынес робота в отдельный класс

Geratoptus committed:
Добавил окно состояния робота

Geratoptus committed:
Исправил поведение робота

MaiklM commented on src/main/java/org/robots/model/Robot.java:
```diff
@@ -0,0 +1,138 @@
+        return !(dist1 > maxCurve) || !(dist2 > maxCurve);
+    }
+
+    private static double distance(double x1, double y1, double x2, double y2) {
```
всю математику стоит выделить в отдельный класс, в роботе она не нужна, это не его ответственность.

MaiklM commented on src/main/java/org/robots/model/Robot.java:
```diff
@@ -0,0 +1,138 @@
+
+        double diff = asNormalizedRadians(angleToTarget - robotDirection);
+
+        if (diff < Math.PI)
```
это в отдельный метод, который настраивает angularVelocity

MaiklM commented on src/main/java/org/robots/model/Robot.java:
```diff
@@ -0,0 +1,138 @@
+import java.util.Observable;
+
+public class Robot extends Observable{
+    private volatile double robotPositionX;
```
positionX, positionY - вполне выносится в отдельный класс point, его вполне можно в качестве target переиспользовать

MaiklM commented on src/main/java/org/robots/model/Robot.java:
```diff
@@ -0,0 +1,138 @@
+        robotPositionY = y;
+    }
+
+    public String getInfo(){
```
а точно ли робот должен знать про то, как показывать свои данные?
Кажется он максимум должен эти данные отдавать, а вот как их в строку преобразовывать - пускай думает класс, которому эти данные передали.

Geratoptus committed:
Доработки

1) Выделил математику игры в отдельный класс
2) Убрал метод getInfo у робота
3) Оформил targetPosition в Point
4) Робот больше не выходит за границы поля

Geratoptus committed:
Доработка границ и robotPosition

Geratoptus committed:
Вынес модель игры в отдельный класс

1) Вынес модель игры в отдельный класс и сменил название соответствующего окна
2) Оформил targetPosition в виде Position
